models/Format.php

```diff
@@ -1239,7 +1187,9 @@ public function insertCardIntoRestrictedlist($card)
             return false; // card not found in database
         }
 
-        if ($this->isCardOnBanList($card) || $this->isCardOnLegalList($card) || $this->isCardOnLegalList($card)) {
+        if ($this->isCardOnRestrictedList($card)) {
+            return true;
+        } elseif ($this->isCardOnBanList($card)) {
```

Comment:
Needs to be `} elseif ($this->isCardOnBanList($card) || $this->isCardOnLegalList($card)) {`

Reply:
done